rpc/personal_api.go

```diff
@@ -75,17 +79,43 @@ func (e *PersonalEthAPI) getKeybaseInfo() ([]keys.Info, error) {
 	return e.cliCtx.Keybase.List()
 }
 
-// ImportRawKey stores the given hex encoded ECDSA key into the key directory,
-// encrypting it with the passphrase.
-// Currently, this is not implemented since the feature is not supported by the keys.
+// ImportRawKey armors and encrypts a given raw hex encoded ECDSA key and stores it into the key directory.
+// The name of the key will have the format "personal_<length-keys>", where <length-keys> is the total number of
+// keys stored on the keyring.
+// NOTE: The key will be both armored and encrypted using the same passphrase.
 func (e *PersonalEthAPI) ImportRawKey(privkey, password string) (common.Address, error) {
-	e.logger.Debug("personal_importRawKey", "error", "not implemented")
-	_, err := crypto.HexToECDSA(privkey)
+	e.logger.Debug("personal_importRawKey")
+	priv, err := crypto.HexToECDSA(privkey)
+	if err != nil {
+		return common.Address{}, err
+	}
+
+	privKey := emintcrypto.PrivKeySecp256k1(crypto.FromECDSA(priv))
+
+	armor := mintkey.EncryptArmorPrivKey(privKey, password, emintcrypto.EthSecp256k1Type)
+
+	// ignore error as we only care about the length of the list
+	list, _ := e.cliCtx.Keybase.List()
+	privKeyName := fmt.Sprintf("personal_%d", len(list))
+
+	if err := e.cliCtx.Keybase.ImportPrivKey(privKeyName, armor, password); err != nil {
+		return common.Address{}, err
+	}
+
+	addr := common.BytesToAddress(privKey.PubKey().Address().Bytes())
+
+	info, err := e.cliCtx.Keybase.Get(privKeyName)
 	if err != nil {
 		return common.Address{}, err
 	}
 
-	return common.Address{}, nil
+	// append key and info to be able to lock and list the account
+	e.keys = append(e.keys, privKey)
```

Comment:
@noot I don't understand why we are using both `e.keys` and `e.ethAPI.keys`?

Reply:
I believe they are identical at the moment, they should probably be consolidated into one. we can probably remove `e.keys` and just use `e.ethAPI.keys`